scalafix-core/shared/src/main/scala/scalafix/internal/rule/DisableSyntax.scala

```diff
@@ -114,17 +98,14 @@ final case class DisableSyntax(
               t.pos
             )
         )
-      case t @ WithMethods(methods)
-          if methods.exists(hasDefaultArgs) && config.noDefaultArgs =>
-        methods
-          .filter(hasDefaultArgs)
-          .map { m =>
-            errorCategory
-              .copy(id = "defaultArgs")
-              .at(
-                "Default args makes it hard to use methods as functions.",
-                m.pos)
-          }
+      case d: Defn.Def if hasDefaultArgs(d) && config.noDefaultArgs =>
```

Comment:
How about matching on `Term.Parameter` and point the caret at the parameter name instead of the def name?

Reply:
We don't want to match default args in class constructor, for example. And I don't want to call `.parent` to check that the default arg is in the `def` :)